sprintMission3/controllers/ArticleController.js

```diff
@@ -0,0 +1,106 @@
+import prisma from '../prisma-client.js';
+
+export const createArticle = async (req, res, next) => {
+  try {
+    const { title, content } = req.body;
+    const newArticle = await prisma.article.create({
+      data: { title, content },
+    });
+    res.status(201).json(newArticle);
+  } catch (error) {
+    next(error); // 에러를 다음 미들웨어(에러 핸들러)로 전달
+  }
+};
+
+// 게시글 목록 조회
+export const getArticles = async (req, res, next) => {
+  try {
+    const page = parseInt(req.query.page || '1');
+    const limit = parseInt(req.query.limit || '10');
+    const search = req.query.search;
+
+    if (page < 1 || limit < 1) {
+      return res.status(400).json({ error: '페이지 설정이 잘못되었습니다.' });
+    }
+
+    const offset = (page - 1) * limit;
+    const whereCondition = search
+      ? {
+          OR: [
+            { title: { contains: search } },
+            { content: { contains: search } },
+          ],
+        }
+      : {};
+
+    const [articles, totalCount] = await prisma.$transaction([
+      prisma.article.findMany({
+        where: whereCondition,
+        select: { id: true, title: true, content: true, createdAt: true },
+        orderBy: { createdAt: 'desc' },
+        skip: offset,
+        take: limit,
+      }),
+      prisma.article.count({ where: whereCondition }),
+    ]);
+
+    const totalPages = Math.ceil(totalCount / limit);
+
+    res.json({
+      data: articles,
+      pagination: { page, limit, totalCount, totalPages },
+    });
+  } catch (error) {
+    next(error);
+  }
+};
+
+// 게시글 상세 조회
+export const getArticleById = async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const article = await prisma.article.findUnique({
+      where: { id: parseInt(id) },
+    });
+    if (article) {
+      res.json(article);
+    } else {
+      res.status(404).json({ error: '게시글을 찾을 수 없습니다.' });
+    }
+  } catch (error) {
+    next(error);
+  }
+};
+
+// 게시글 수정
+export const updateArticle = async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const { title, content } = req.body;
+
+    const dataToUpdate = {};
+    if (title !== undefined) dataToUpdate.title = title;

```

Comment:
반복되어 나타나는 코드는 다음과 같이 모듈화 하면 좋아요. 이런식으로 조금씩 모듈화해보는 습관을 길러봐주세요.


```suggestion
export const isUndefined(value) { return value === undefined }

export const isNotUndefined(value) { return isUndefined(value) === false }

// ...
    if (isNotUndefined(title)) dataToUpdate.title = title;
```